File: app/src/main/java/com/hjc/reader/viewmodel/wan/NavigationViewModel.java
package com.hjc.reader.viewmodel.wan;


import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.MutableLiveData;

import com.blankj.utilcode.util.ToastUtils;
import com.hjc.baselib.http.RxSchedulers;
import com.hjc.baselib.viewmodel.CommonViewModel;
import com.hjc.reader.bean.response.WanNavigationBean;
import com.hjc.reader.http.RetrofitHelper;
import com.hjc.reader.http.observer.BasePageObserver;

import java.util.List;

public class NavigationViewModel extends CommonViewModel {

    // 左边导航列表数据
    private MutableLiveData<List<WanNavigationBean.DataBean>> navigationLiveData = new MutableLiveData<>();

    //右边内容列表数据
    private MutableLiveData<List<WanNavigationBean.DataBean>> navigationContentLiveData = new MutableLiveData<>();

    public NavigationViewModel(@NonNull Application application) {
        super(application);
    }


    /**
     * 获取导航列表数据
     *
     * @param isFirst 是否第一次加载
     */
    public void loadNavigationList(boolean isFirst) {
        RetrofitHelper.getInstance().getWanAndroidService()
                .getNavigationList()
                .compose(RxSchedulers.ioToMain())
                .subscribe(new BasePageObserver<WanNavigationBean>(this, isFirst) {
                    @Override
                    public void onSuccess(WanNavigationBean result) {
                        if (result.getErrorCode() == 0) {
                            List<WanNavigationBean.DataBean> dataList = result.getData();
                            if (dataList != null && dataList.size() > 0) {
                                navigationLiveData.setValue(dataList);
                                navigationContentLiveData.setValue(dataList);
                            }
                        } else {
                            ToastUtils.showShort(result.getErrorMsg());
                        }
                    }
                });
    }

    public MutableLiveData<List<WanNavigationBean.DataBean>> getNavigationLiveData() {
        return navigationLiveData;
    }

    public MutableLiveData<List<WanNavigationBean.DataBean>> getNavigationContentLiveData() {
        return navigationContentLiveData;
    }

}

File: module-wan/src/main/java/com/hjc/learn/wan/viewmodel/NavigationViewModel.kt
package com.hjc.learn.wan.viewmodel

import android.app.Application
import androidx.lifecycle.MutableLiveData
import com.hjc.library_common.viewmodel.KotlinViewModel
import com.hjc.library_net.RetrofitClient
import com.hjc.library_net.model.WanNavigationBean
import java.net.SocketTimeoutException
import java.net.UnknownHostException

/**
 * @Author: HJC
 * @Date: 2021/2/23 9:36
 * @Description: 导航数据ViewModel
 */
class NavigationViewModel(application: Application) : KotlinViewModel(application) {

    // 左边导航列表数据
    val navigationData = MutableLiveData<MutableList<WanNavigationBean>>()

    //右边内容列表数据
    val navigationContentData = MutableLiveData<MutableList<WanNavigationBean>>()

    /**
     * 获取导航列表数据
     *
     * @param isFirst 是否第一次加载
     */
    fun loadNavigationList(isFirst: Boolean) {
        launchWan({ RetrofitClient.getApiService1().getNavigationList() }, { result ->
            result?.let {
                if (it.size > 0) {
                    showContent()
                    navigationData.value = it
                    navigationContentData.value = it
                } else {
                    showEmpty()
                }
            }
        }, { e ->
            if (e is UnknownHostException || e is SocketTimeoutException) {
                showError()
            } else {
                showTimeout()
            }
        }, isShowProgress = isFirst)
    }
}